### PR TITLE
Fix test framework's groupby expression support

### LINKED ISSUE #5418: Executor test framework failed to support aggregation without groupby expr
## Bug Report

Current test framework will throw exception when an aggregation query does not have groupby expression. e.g. `select Max(a) from t1`.

And the problem lies in `DAGRequestBuilder::aggregation(ASTPtr agg_func, ASTPtr group_by_expr)`

```cpp
DAGRequestBuilder & DAGRequestBuilder::aggregation(ASTPtr agg_func, ASTPtr group_by_expr)
{
    auto agg_funcs = std::make_shared<ASTExpressionList>();
    auto group_by_exprs = std::make_shared<ASTExpressionList>();
    agg_funcs->children.push_back(agg_func);
    group_by_exprs->children.push_back(group_by_expr);
    return buildAggregation(agg_funcs, group_by_exprs);
}
```

When `agg_func` or `group_by_expr` is nullptr, the nullptr will be pushed back into `ASTExpressionList`.

### 1. Minimal reproduce step (Required)

By writing a simple agg test like following

```cpp
TEST_F(JoinExecutorTestRunner, agg)
try
{
    context.addMockTable("join_agg", "t1", {{"a", TiDB::TP::TypeLong}, {"b", TiDB::TP::TypeLong}}, {toVec<Int32>("a", {1, 1, 3, 4}), toVec<Int32>("b", {1, 1, 4, 1})});
    auto request = context.scan("join_agg", "t1")
        .aggregation({Max(col("a"))}, {})
        .build(context);
    
    {
        ASSERT_COLUMNS_EQ_UR(createColumns({toNullableVec<Int32>({4})}), executeStreams(request));
    }
}
CATCH
```

### 2. What did you expect to see? (Required)

test passed

### 3. What did you see instead (Required)

```cpp
unknown file: Failure
C++ exception with description "std::bad_typeid" thrown in the test body.
```

### 4. What is your TiFlash version? (Required)

master

### PR DESCRIPTION
**Pull request by Willendless**
### What problem does this PR solve?

Issue Number: close #5418 

Problem Summary:

### What is changed and how it works?

+ check whether `agg/groupby` ASTPtr is null before pushing them into expression list 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/tests/gtest_executor.cpp [hunk 1]
```diff
@@ -60,5 +60,29 @@ try
 }
 CATCH
 
+// todo: move this test to gtest_aggregation_executor.cpp
```
Comment (SeaRise) [inline dbms/src/Flash/tests/gtest_executor.cpp:63]: maybe you can do now?

**Review by SeaRise — APPROVED** [review]
LGTM

File: dbms/src/Flash/tests/gtest_executor.cpp (hunk 1 above)
Comment (Willendless) [reply dbms/src/Flash/tests/gtest_executor.cpp:63]: Sure

**Review by ywqzzy — APPROVED** [review]
LGTM

**Comment by Willendless** [comment]
/cc @xzhangxian1008 

feel free to modify my test case when you support more aggregation expressions and add corresponding tests

:)

**Pull request merged**